packages/headless/src/composables/menuButton/createMenuButton.ts

```diff
@@ -0,0 +1,72 @@
+import { computed, ref } from "vue";
+import { createBuilder } from "../../utils/builder";
+import { createId } from "../../utils/id";
+
+/**
+ * Purpose of the onSelect function is to set selected option (page) as active
+ */
+export type CreateMenuButtonOptions = { onSelect: (value: string) => void };
+
+export const createMenuButton = createBuilder((options: CreateMenuButtonOptions) => {
+  const menuId = createId("menu");
+  const buttonId = createId("menu-button");
+  const isExpanded = ref<boolean>(false);
+  let timeout: ReturnType<typeof setTimeout> | undefined;
+
+  /**
+   * Debounced hodden state that will only be toggled after a given timeout.
+   */
+  const debouncedHidden = computed({
+    get: () => isExpanded.value,
+    set: (newValue) => {
+      clearTimeout(timeout);
+      timeout = setTimeout(() => {
+        isExpanded.value = newValue;
+      }, 400);
+    },
+  });
+
+  const hoverEvents = computed(() => {
+    return {
+      onMouseover: () => (debouncedHidden.value = true),
+      onMouseout: () => (debouncedHidden.value = false),
+      onFocusin: () => (isExpanded.value = true),
+      onFocusout: () => (isExpanded.value = false),
+    };
+  });
```

Comment:
```suggestion
import { debounce } from "../../utils/timer";

  /**
   * Debounced expanded state that will only be toggled after a given timeout.
   */
  const updateDebouncedExpanded = debounce(
    (expanded: boolean) => (isExpanded.value = expanded),
    400,
  );

  const hoverEvents = computed(() => {
    return {
      onMouseover: () => updateDebouncedExpanded(true),
      onMouseout: () => updateDebouncedExpanded(false),
      onFocusin: () => (isExpanded.value = true),
      onFocusout: () => (isExpanded.value = false),
    };
  });
```

We already have a debounce utility that we can make use of here :)

Reply:
Also, should we only use the debounce for closing? Otherwise it might feel weird if it does not open when hovering.

Or maybe change the timeout to 200 or so

Reply:
Now I'm using it only on closing